diff --git a/device.go b/device.go
index e324fa4..76ddfbd 100644
--- a/device.go
+++ b/device.go
@@ -82,7 +82,7 @@ func getRplidarDevice(devicePath string) (*rplidarDevice, error) {
 	serialNum := devInfo.GetSerialnum()
 	var serialNumStr string
 	for pos := 0; pos < 16; pos++ {
-		serialNumStr += fmt.Sprintf("%02X", gen.ByteArray_getitem(serialNum, pos))
+		serialNumStr += fmt.Sprintf("%02X", gen.ByteArray_getitem(serialNum, int64(pos)))
 	}
 
 	firmwareVer := fmt.Sprintf("%d.%02d",
diff --git a/rplidar.go b/rplidar.go
index a0ea2e7..f599f84 100644
--- a/rplidar.go
+++ b/rplidar.go
@@ -195,7 +195,7 @@ func (rp *rplidar) scan(ctx context.Context, numScans int) (pointcloud.PointClou
 
 		for pos := 0; pos < int(nodeCount); pos++ {
 
-			node := gen.MeasurementNodeHqArray_getitem(rp.nodes, pos)
+			node := gen.MeasurementNodeHqArray_getitem(rp.nodes, int64(pos))
 
 			if node.GetDist_mm_q2() == 0 {
 				dropCount++
